### ISSUE TITLE
make the API key setting user-friendlier

### ISSUE THREAD

**Issue by maelle** [ropensci/opencage#51]
mostly stealing all ideas from https://github.com/hrbrmstr/omdbapi/blob/3b69849a540c844a09cf03fb4005b1fd16f80eb2/R/omdb_api_key.R

**Comment by dpprdan** [comment]
This sets the API key only for the current session, or am I missing something?

**Comment by maelle** [comment]
No since it uses `Sys.setenv`?

**Comment by dpprdan** [comment]
Yeah that's what I thought. For some reason I had the impression that this would set the key permanently when I glanced at it the first time. Don't really know why, though. 
To paraphrase [you](https://github.com/ropensci/opencage/issues/32#issuecomment-370234195): In the vignette documenting this we can show code example and mention `usethis::edit_r_environ()` as a way to easily find where the .Renviron is/open it. And link to [Efficient R Programming](https://csgillespie.github.io/efficientR/set-up.html#renviron).

**Comment by maelle** [comment]
I'm still torn between documenting it or making it even easier 😂

**Comment by dpprdan** [comment]
Just leaving this here as a note: One possible solution would be to `Sys.setenv(OPENCAGE_KEY)` with `oc_configure` per session. Also for folks using e.g. the keyring pkg instead of `.Renviron`.

**Comment by maelle** [comment]
So your first idea would mean restarting once per session?

For further ref link to keyring https://github.com/r-lib/keyring

**Comment by dpprdan** [comment]
I meant something like this: 

```r
oc_configure <-
  function(
    key = NULL,
    max_rate_per_sec =
      getOption("oc_max_rate_per_sec", default = 1L),
    ...
  ) {
    if (!is.null(key)) {
      Sys.setenv("OPENCAGE_KEY" = key)
    }
    ratelimitr::UPDATE_RATE(
      oc_get_limited,
      ratelimitr::rate(n = max_rate_per_sec, period = 1L)
    )
  }
```

Probably needs a `message()` (or check + `force` argument) in case `"OPENCAGE_KEY"` is already set in `.Renviron`.

I like Bob's take on `omdb_api_key()`. One function setting and getting a value at the same time makes me a bit uneasy, but for this use-case it's quite elegant. However, AFAIK his solution does not support the use case where you would want to permanently store your key with e.g. [keyring](https://github.com/r-lib/keyring) (or [secret](https://github.com/gaborcsardi/secret), which seems the way to go for company accounts) and not in `.Renviron`. With the above, one could do this at the beginning of the session/script:

```r
oc_configure(key = keyring::key_get("opencage"))
```

instead of providing the key with every call like this: 

```r
oc_forward("something,something", key = keyring::key_get("opencage")) 
```

**Comment by maelle** [comment]
Or we could more simply message about `usethis::edit_r_environ()` and other possible solutions 🤔

**Comment by dpprdan** [comment]
You seem to be [torn](https://github.com/ropensci/opencage/issues/51#issuecomment-370243598), indeed. 😉

**Comment by dpprdan** [comment]
FWIW, openrouteservice-r is (planning on) moving away from keyring to `.Renviron`: https://github.com/GIScience/openrouteservice-r/issues/34

**Comment by maelle** [comment]
Ah that's interesting! But I thought that keyring also supported `.Renviron`, I'd need to look into its settings a bit more.

**Comment by dpprdan** [comment]
@maelle is this is closed by #99?

**Comment by maelle** [comment]
Yep, indeed!
